### DISCUSSION TITLE
Cannot make atom typesafe it shows as any

### DISCUSSION THREAD

**Discussion by khanakia** [pmndrs/jotai#2591] [Q&A]
I have this code and when i mouse hover to `immerSetup.screen` in vscode it shows any

How do I make it typesafe ?

```ts
type Setup = {
  screen: string;
};

const defaultSetup = {
  screen: "step1",
};

// Original atom.
const setupImmerAtom = atomWithImmer<Setup>(defaultSetup);

const setScreen = (screen: Screen) => {
  const [setupAtom, setAtomSetup] = useAtom(setupImmerAtom);
  setAtomSetup((immerSetup) => {
    immerSetup.screen = screen;
    return immerSetup;
  });
};
```

![Screenshot 2024-06-01 at 7 07 33 PM](https://github.com/pmndrs/jotai/assets/2174170/1faf20c5-1550-49ce-85f9-3807d4a1da71)

**Comment by rothsandro** [comment]
Works fine for me https://tsplay.dev/wRby1w

Can you create a reproduction?

**Comment by khanakia** [comment]
Now it started working fine after reloading, which is strange. I am using remix with vs code.
